### PR TITLE
CLN: remove fill_tuple kludge

### PR DESCRIPTION
**Pull request by jbrockmendel**

### CODE REVIEW

**Review by jorisvandenbossche — APPROVED** [review]

File: pandas/core/internals/managers.py [hunk 1]
```diff
@@ -1396,7 +1394,7 @@ def _slice_take_blocks_ax0(self, slice_or_indexer, fill_tuple=None):
                             blklocs[mgr_locs.indexer],
                             axis=0,
                             new_mgr_locs=mgr_locs,
-                            fill_tuple=None,
+                            fill_value=lib.no_default,
```
Comment (simonjayhawkins) [inline pandas/core/internals/managers.py:1397]: should this be removed? isn't it an anti-pattern to explicitly pass this.

File: pandas/core/internals/managers.py [hunk 2]
```diff
@@ -1339,16 +1339,15 @@ def _slice_take_blocks_ax0(self, slice_or_indexer, fill_tuple=None):
             if sl_type in ("slice", "mask"):
                 return [blk.getitem_block(slobj, new_mgr_locs=slice(0, sllen))]
             elif not allow_fill or self.ndim == 1:
-                if allow_fill and fill_tuple[0] is None:
+                if allow_fill and fill_value is None:
```
Comment (simonjayhawkins) [inline pandas/core/internals/managers.py:1342]: is allow_fill now redundant here?

**Review by simonjayhawkins — APPROVED** [review]
Thanks @jbrockmendel generally lgtm.

File: pandas/core/internals/managers.py (hunk 2 above)
Comment (jreback) [reply pandas/core/internals/managers.py:1342]: allow_fill is tricky to remove (but this is a start to it)

Comment (jorisvandenbossche) [reply pandas/core/internals/managers.py:1342]: The `allow_fill` is from checking `fill_value is not lib.default`. So `fill_value is None` is still a different check I think.

Comment (jbrockmendel) [reply pandas/core/internals/managers.py:1342]: what @jorisvandenbossche said

File: pandas/core/internals/managers.py (hunk 1 above)
Comment (jbrockmendel) [reply pandas/core/internals/managers.py:1397]: good idea

File: pandas/core/internals/managers.py (hunk 2 above)
Comment (simonjayhawkins) [reply pandas/core/internals/managers.py:1342]: if `fill_value is None`, isn't `allow_fill`, by definition, `True`. so `allow_fill` is redundant here and `if fill_value is None:` would be the same? nbd though, i must be missing something. will look again. We no longer need the guard, that was previously required with the `fill_tuple` \_\_getitem__ call.

**Pull request merged**